### ISSUE TITLE
Bug: pdfminer.six fails to read certain japanese fonts and returns cid value

### ISSUE THREAD

**Issue by Kurinosuke118** [pdfminer/pdfminer.six#927]
**Bug report**

`pdfminer.six` fails to read certain japanese fonts and returns `cid` value. 
I assume this is caused by pdfminer's `CMap` not being able to convert the cid code of a particular Japanese font to a character code.

This bug occurs when parsing Japanese PDF files with the following code.
```
from pdfminer.high_level import extract_text

text = extract_text("example.pdf")
print(text)
```

**Comment**
I think that the file contents in [1] need to update with the file contents in [2] and update a `CMap` with `conv_cmap.py` in the `pdfminer.six` repository.
[1] https://github.com/pdfminer/pdfminer.six/blob/master/cmaprsrc/cid2code_Adobe_Japan1.txt 
[2] https://github.com/adobe-type-tools/cmap-resources/tree/master/Adobe-Japan1-7

**Comment by pietermarsman** [comment]
Thanks for the tip! Do you happen to know if Japan1-7 is a superset of Japan1? In other words, that we are fully backwards compatible if we switch from Japan1 to Japan1-7?

**Comment by Kurinosuke118** [comment]
@pietermarsman Thanks for the reply!
> Do you happen to know if Japan1-7 is a superset of Japan1?

I think that `cid2code_Adobe_Japan1.txt` is a superset of `Adobe-Japan1-7/cid2code.txt`.
When I rewrite the contents of the `cid2code_Adobe_Japan1.txt` file to `Adobe-Japan1-7/cid2code.txt`, need to create a `*.pickle.gz` file? Also, tell me how to create the `*.pickle.gz` file.

**Comment by pietermarsman** [comment]
I'm not an expert on this myself, the files were already there when I first ran into pdfminer. But I can give you some directions: 

- The [Makefile](https://github.com/pdfminer/pdfminer.six/blob/master/Makefile) has a couple of commands to build the pickle files
- It uses the [tools/conv_cmap.py](https://github.com/pdfminer/pdfminer.six/blob/master/tools/conv_cmap.py) file to do the conversion
- You probably want to add/change one of the cmaps in [cmapsrc](https://github.com/pdfminer/pdfminer.six/tree/master/cmaprsrc).

**Comment by exander77** [comment]
> **Bug report**
> 
> `pdfminer.six` fails to read certain japanese fonts and returns `cid` value (cid:xxx). I think this is caused by pdfminer's `CMap` not being able to convert the cid code of a particular Japanese font to a character code.
> 
> **To Reproduce** This bug occurs when the following Japanese PDF file is parsed with the following code.
> 
> PDF file 　https://www.maff.go.jp/j/tokei/kouhyou/naisui_gyosei/attach/pdf/index-15.pdf
> 
> Code
> 
> ```
> from pdfminer.high_level import extract_text
> 
> text = extract_text("xxx.pdf")
> print(text)
> ```
> 
> Ouput
> 
> ```
> ...
> (cid:1748)(cid:10492)年(cid:2233)(cid:2286)(cid:2225)
> 
> 年(cid:2057)(cid:7711)
> ...
> ```
> 
> **Comment**
> 
> I think that the file contents in [1] need to update with the file contents in [2] and update a `CMap` with `tools/conv_cmap.py` in the `pdfminer.six` repository.
> 
> [1] https://github.com/pdfminer/pdfminer.six/blob/master/cmaprsrc/cid2code_Adobe_Japan1.txt [2] https://github.com/adobe-type-tools/cmap-resources/blob/master/Adobe-Japan1-7/cid2code.txt

Is the original document available?

**Comment by dhdaines** [comment]
Yes, it might be possible to fix this by updating the cmap files with the most recent ones from the public github repository (and this would be good, because the links in pdfminer/cmap/README.txt don't work anymore).

If the original reporter is still here - can you try running [playa --text](https://github.com/dhdaines/playa) on your PDF to see if it extracts the text (it won't return cid values, though, just empty strings)?  I'll port the fix from there to pdfminer.six.

**Comment by dhdaines** [comment]
Can you also try #1257 to see if it helps you?
